qiskit_ibm_runtime/execution_span.py

```diff
@@ -0,0 +1,94 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Execution span classes."""
+
+from typing import Union, Iterable, Sequence, TypeVar
+from datetime import datetime
+from dataclasses import dataclass
+
+
+"""The format accepted by ``numpy.ndarray.__getitem__()``."""
+SliceType = tuple[Union[slice, int, list[int]], ...]
+ExecutionSpanT = TypeVar("ExecutionSpanT", bound="ExecutionSpan")
+
+
+@dataclass(frozen=True, slots=True)
+class ExecutionSpan:
+    """Stores an execution time span for a subset of job data."""
+
+    start: datetime
+    """The start time of the span, in UTC."""
+
+    stop: datetime
+    """The stop time of the span, in UTC."""
+
+    data_slices: dict[int, SliceType]
+    r"""Which data have dependence on this execution span.
+
+    Data from the primitives are array-based, with every field in a 
+    :class:`~PubResult`\s :class:`~.DataBin` sharing the same base shape.
+    Therefore, the format of this field is a mapping from pub indexes to 
+    the same format accepted by 
+    NumPy slicing, where each value indicates which slice of each field in the
+    data bin depend on raw data collected during this execution span.
+    """
+
+    @property
+    def duration(self) -> float:
+        return (self.stop - self.start).seconds
```

Comment:
I think what we want here is `total_seconds()` and not `seconds`:
```
>>> from datetime import datetime
>>> t1 = datetime.now()
>>> t2 = datetime.now()
>>> t2 - t1
datetime.timedelta(seconds=15, microseconds=241326)
>>> (t2 - t1).seconds
15
>>> (t2 - t1).total_seconds
<built-in method total_seconds of datetime.timedelta object at 0x7f81ce98b1b0>
>>> (t2 - t1).total_seconds()
15.241326
```

Reply:
Thanks, I wrote that code blind and didn't check the `time` docs.